### PR TITLE
Add missing counter operations

### PR DESCRIPTION
**Pull request by davidor**
This PR adds the operations that were missing for `Counter`: get_config, decrement, reset, compare_and_set, and compare_and_swap.

With this we already have all the operations documented in https://infinispan.org/docs/stable/titles/rest/rest.html#rest_v2_counters , although we don't support all the options they accept yet.

### CODE REVIEW

File: tests/counters.rs [hunk 1]
```diff
@@ -71,6 +72,27 @@ mod counters {
...
+        let counter_name = "test_counter";
+        let initial_val = 10;
+
+        let _ = run(&counters::create_strong(counter_name).with_value(initial_val)).await;
+        let resp = run(&counters::get_config(counter_name)).await;
+
+        assert!(resp.status().is_success());
+
+        let config_bytes = resp.bytes().await.unwrap();
+        let config_str = std::str::from_utf8(&config_bytes).unwrap();
```
Comment (unleashed) [inline tests/counters.rs:89]: I think you can use here `str::from_utf8` without `std::` since it should be `use`d implicitly via the prelude.

**Review by unleashed — APPROVED** [review]
Looks good - check comment.

File: tests/counters.rs (hunk 1 above)
Comment (davidor) [reply tests/counters.rs:89]: No, I tried it and it doesn't work.

Comment (unleashed) [reply tests/counters.rs:89]: Then use `core::str` for maximum future-proofing.

**Pull request merged**